sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -268,11 +273,20 @@ def _close_consumer(self, partition_id, consumer, reason):
         )
 
         if self._partition_close_handler:
-            self._handle_callback(
-                self._partition_close_handler,
-                self._partition_contexts[partition_id],
-                reason,
-            )
+            try:
+                self._partition_close_handler(self._partition_contexts[partition_id], reason)
+            except Exception as err:  # pylint:disable=broad-except
+                _LOGGER.warning(
```

Comment:
We're already logging a warning in `def _process_error` so this will log double warnings. Was that intended?

Reply:
should improve this